### DISCUSSION TITLE
[Prettier Plugin] Allow wildcards for tailwindAttributes array

### DISCUSSION THREAD

**Discussion by reintroducing** [tailwindlabs/tailwindcss#11838] [Ideas]
I'm using tailwind in a React UI library where I expose different className properties to provide overwritable props on core components. For instance, a Table component may have the following props:
```ts
/** Additional class(es) to add to the root element. */
classNameRoot?: string;
/** Additional class(es) to add to the table element. */
classNameTable?: string;
/** Additional class(es) to add to the table head element. */
classNameHead?: string;
/** Additional class(es) to add to each table head cell element. */
classNameHeadCell?: string;
/** Additional class(es) to add to the table body element. */
classNameBody?: string;
/** Additional class(es) to add to every table row. */
classNameRow?: string;
/** Additional class(es) to add to every table body cell. */
classNameBodyCell?: string;
/** Additional class(es) to add to the table body element when there is no data available. */
classNameNoData?: string;
/** Additional class(es) to add to the table element when highlightOnHover is true. */
classNameHighlighted?: string;
/** Additional class(es) to add to the container element when isLoading is true. */
classNameLoading?: string;
/** Additional class(es) to add to the pagination element. */
classNamePagination?: string;
/** Additional class(es) to add to the header element. */
classNameHeader?: string;
/** Additional class(es) to add to the footer element. */
classNameFooter?: string;
```

As you can see, when I create a new `<Table />` I would be passing no `className` prop, rather (potentially) multiple different props that start with `className[something]`. Is there a way to provide something like this in the config so that the classes passed to these props can be auto sorted? Something like the following would be great: `tailwindAttributes: ['className*'],` but I don't know if this already exists and I'm just not using the proper syntax?

Thank you.

**Comment by CombeeMike** [comment]
Also looking for this. Did you ever find a solution?

**Reply by reintroducing** [reply]
Unfortunately no and this has gotten no traction :/

**Reply by pixelchutes** [reply]
Would also love to see consideration for wildcard support e.g. `className*`, `*ClassName`, etc.

^ This would definitely simplify the Prettier config (`tailwindAttributes` setting), while also better supporting future use cases as the codebase grows.

In the current state, a fraction of our project benefits from this very helpful plugin, while other parts are inadvertently missed.
